### PR TITLE
Install oci8 using managed way

### PR DESCRIPTION
**Pull request by mvorisek**
thanks to:
https://github.com/mlocati/docker-php-extension-installer/pull/215
https://github.com/mlocati/docker-php-extension-installer/pull/220
https://github.com/mlocati/docker-php-extension-installer/pull/234

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -36,24 +36,10 @@ RUN apk add libstdc++ unixodbc unixodbc-dev && \
...
-RUN apk add libaio libnsl libc6-compat && \
-    docker-php-ext-configure pdo_oci --with-pdo-oci=instantclient,/usr/local/instantclient && \
-    ln -s /usr/lib/libnsl.so.2 /usr/lib/libnsl.so.1 && \
-    ln -s /lib/libc.so.6 /usr/lib/libresolv.so.2 && \
-    ln -s /lib64/ld-linux-x86-64.so.2 /usr/lib/ld-linux-x86-64.so.2 && \
-    docker-php-ext-install pdo_oci
+# install Oracle Instant client & pdo_oci PHP extension
+COPY --from=mlocati/php-extension-installer /usr/bin/install-php-extensions /usr/local/bin/
+RUN install-php-extensions oci8
+RUN docker-php-ext-install pdo_oci
```
Comment (mvorisek) [inline Dockerfile.template]: @mlocati can support for `pdo_oci` be easily added?

Comment (mlocati) [reply Dockerfile.template]: @mvorisek Sure! See https://github.com/mlocati/docker-php-extension-installer/pull/234

**Review by romaninsh — APPROVED** [review]

**Comment by mlocati** [comment]
Out of curiosity, is there any reason you don't use php-extension-installer to install all the extensions?

**Comment by mvorisek** [comment]
> 
> 
> Out of curiosity, is there any reason you don't use php-extension-installer to install all the extensions?

👍 for asking for feedback. Starting with early PHP 8.0 alpha we required images for PHP 7.2 - 8.0 with a lot of extensions and there was no managed way to install severals exts (xdebug, oci8, sqlsrv, ...) so we ended up with `docker-php-ext-install` (because it feels the most native way for php Docker images), `pecl`, `pickle` + a few manuals builds (as there was really no other clean way).

But you are right, we should preferably replace `pecl`/`pickle` with your `php-extension-installer` as deps do not have to be installed manually.

For now, I would like to enable `pdo_sqlsrv` for PHP 8.0. For PHP 7.x it is installed like: https://github.com/mvorisek/image/blob/facd024d7767460dca29a1c1248cd7567c5cf8d5/Dockerfile.template#L35 but for PHP 8.0 there is no clean way, see https://github.com/mvorisek/image/blob/facd024d7767460dca29a1c1248cd7567c5cf8d5/8.0/Dockerfile#L40 , as there is some issue with sqlsrv ext beta version format. Do you think https://github.com/microsoft/msphpsql `5.9.0-beta2` can be supported by `php-extension-installer` until a stable release for PHP 8.0 is available? Is it true, that manuall ODBC drivers install https://github.com/mvorisek/image/blob/facd024d7767460dca29a1c1248cd7567c5cf8d5/Dockerfile.template#L28 can be dropped then?

Also, it will be nice if `pickle` can be installed with `php-extension-installer` like composer can be.

**Comment by mlocati** [comment]
Well, I've never used the SQL Server extensions. so I don't know if the ODBC drivers are required.
But let me say that the idea behind my docker-php-extension-installer project is not that I'm the Source Of Truth&trade;: I like everybody else, don't know every PHP extensions :wink:

I created that project in the hope to have a common project where everybody else can contribute to.

For example, you need xdebug for PHP 8. And other thousands of users need it too.
So instead of having thousands of people trying to figure out how to install xdebug with PHP 8, in my opinion people could contribute to docker-php-extension-installer: that way we'd have a simple solution for everyone.

For that reason I'm trying to be as fast as possible giving feedback to issues and pull requests, and the release cycle of that project is very fast (I find it so tedious when I have to wait for weeks before seeing my contributions to other projects being reviewed...).

So, this is my suggestion: instead of experimenting installing PHP extensions in your project, what about submitting pull requests to docker-php-extension-installer? It has a pretty good testing system, and you'd help others too. Also, others may contribute to your changes, improving everything. It's a win-win approach.

**Pull request merged**